modules/geo-layers/src/mesh-layer/mesh-layer.js

```diff
@@ -98,15 +102,17 @@ export default class _MeshLayer extends SimpleMeshLayer {
     const unlit = Boolean(
       pbrMaterial.pbrMetallicRoughness && pbrMaterial.pbrMetallicRoughness.baseColorTexture
     );
-    const materialParser = new GLTFMaterialParser(this.context.gl, {
+
+    this.destroyMaterialParserTextures();
```

Comment:
This helper method changes `parseMaterial()` from a pure function to something that mutates the layer state. Either set the new state here or just simply call `this.state.materialParser?.delete()`. Generally speaking it's a good idea not to rely on the caller to maintain `state.materialParser` when this method relies on it.

Reply:
Ok, I'v kept only `this.state.materialParser?.delete()` to avoid mutation.

Reply:
Can you just remove the `destroyMaterialParserTextures` method and incorporate it in to `finalizeState`? It seems unnecessary to have a helper function for two lines of code that is only called once...

Reply:
Done